openvdb/openvdb/openvdb.cc

```diff
@@ -30,20 +33,21 @@ namespace openvdb {
 OPENVDB_USE_VERSION_NAMESPACE
 namespace OPENVDB_VERSION_NAME {
 
-typedef tbb::mutex Mutex;
-typedef Mutex::scoped_lock Lock;
+using Mutex = std::mutex;
+using Lock = std::lock_guard<Mutex>;
 
 namespace {
 // Declare this at file scope to ensure thread-safe initialization.
 Mutex sInitMutex;
-bool sIsInitialized = false;
+std::atomic<bool> sIsInitialized{false};
 }
 
 void
 initialize()
 {
+    if (sIsInitialized.load(std::memory_order_acquire)) return;
```

Comment:
I think (?) we want at least acquire semantics, otherwise we could have an out-of-date read (right?) and decide that it's initialized when it's in fact no longer initialized? (Although tearing down seems sketchy to me.)

Reply:
The only approach here that I'd feel confident in that would still permit `uninitialize` would be to have the shared state that goes with initialization be stored attached to a in a `static std::shared_ptr` and have initialization atomically get it, check it and return a `std::shared_ptr` to it, making it the caller's job to keep the `shared_ptr` alive.

As it is, I think the caller of `uninitialize` always has to worry about if anyone is still using things. If `uninitialize` weren't needed, we could just use `std::once_flag`.

Reply:
This is what I meant about uninitialize being sketchy :>

Acquire doesn't stop you from having an out-of-date read.  You can still have an out of date read as there is a window between when you test this and the double check lock.  So you can read this with acquire; properly validate that it is initialized;  have else-thread unintialize; and then return as if initialized.

As soon as you have uninitialize interleaving with initialize this is all quite broken, as there is no sense of a stack.  So we might as well make the check as fast as possible :>

Reply:
True. I guess I'm just thinking that with full-relaxed loading, it could have been `uninitialize`d a moment ago and we would incorrectly conclude that it's initialized. With acquire semantics it's still suspect, but at least a reasonable case should be correct. I'm not changing it to relaxed in my PR :-)